### PR TITLE
background sync

### PR DESCRIPTION
**Pull request by Anna812**
Setup alarm manager
Is enabled/disabled from Settings
BroadcastReceiver updates feed, and sends Event on Eventbus
Feedfragment is Subscribed to this event, and updates listview

### CODE REVIEW

File: app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/FeedFragment.java [hunk 1]
```diff
@@ -96,12 +92,14 @@ public void showNewsItems() {
         call.enqueue(new Callback<ArrayList<NewsItem>>() {
             @Override
             public void onResponse(Call<ArrayList<NewsItem>> call, Response<ArrayList<NewsItem>> response) {
+                if(tabNumber == 1) {
+                    news = response.body();
+                }
                 adapter.addAll(response.body());
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/FeedFragment.java]: @Anna812 why not just the 98th line? why the need to store them in a variable?

File: app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/SettingsActivity.java [hunk 2]
```diff
@@ -26,6 +31,35 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                PendingIntent.FLAG_UPDATE_CURRENT);
+
+        AlarmManager alarm = (AlarmManager) this.getSystemService(Context.ALARM_SERVICE);
+        alarm.setInexactRepeating(AlarmManager.RTC_WAKEUP, System.currentTimeMillis(), 10000L, pIntent);
+    }
+
+    public void cancelAlarm() {
+        Intent intent = new Intent(this, BackgroundSyncReceiver.class);
+        final PendingIntent pIntent = PendingIntent.getBroadcast(this, 0, intent,
+                PendingIntent.FLAG_UPDATE_CURRENT);
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/SettingsActivity.java]: @Anna812 refactor this to a method and use it in the schedule as well

File: app/src/main/java/com/greenfox/fuchsit/zerdareader/backgroundSync/BackgroundSyncReceiver.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    private void updateNewsItems(Context context) {
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        DaggerMockServerComponent.builder().build().inject(this);
+
+        Call<ArrayList<NewsItem>> call = apiService.getNewsItems(sharedPreferences.getString("token", ""));
+        call.enqueue(new Callback<ArrayList<NewsItem>>() {
+            @Override
+            public void onResponse(Call<ArrayList<NewsItem>> call, Response<ArrayList<NewsItem>> response) {
+                news = response.body();
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/fuchsit/zerdareader/backgroundSync/BackgroundSyncReceiver.java]: @Anna812 Eventbus posting should be here.

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/FeedFragment.java (hunk 1 above)
Comment (Anna812) [reply app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/FeedFragment.java]: no idea, but deleted unnecessary lines

File: app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/SettingsActivity.java (hunk 2 above)
Comment (Anna812) [reply app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/SettingsActivity.java]: done

File: app/src/main/java/com/greenfox/fuchsit/zerdareader/backgroundSync/BackgroundSyncReceiver.java (hunk 3 above)
Comment (Anna812) [reply app/src/main/java/com/greenfox/fuchsit/zerdareader/backgroundSync/BackgroundSyncReceiver.java]: done

**Review by adamgyulavari — APPROVED** [review]

**Comment by adamgyulavari** [comment]
@Anna812 resolve conflicts

**Comment by Anna812** [comment]
resolved conflicts with master

**Review by adamgyulavari — APPROVED** [review]

**Pull request merged**